array/array.go

```diff
@@ -0,0 +1,144 @@
+package array
+
+import (
+	"errors"
+)
+
+const (
+	// EmptyValue is a constant indicates that
+	// a value returned from the Array is empty.
+	EmptyValue       = 0
```

Comment:
Default value should be a value which is rarely used. I suggest to use min value of `int` type instead of 0.

Reply:
I remember @thanks173 gave a review saying it can be 0, because 0 is a falsey value.